src/variables.ts

```diff
@@ -2,22 +2,35 @@ import { GitHubQuery } from './types';
 import { TemplateSrv } from '@grafana/runtime';
 import { ScopedVars } from '@grafana/data';
 
-export const replaceVariable = (t: TemplateSrv, value?: string, scoped: ScopedVars): string | undefined => {
-  return !!value ? t.replace(value, scoped) : value;
+export const replaceVariable = (
+  t: TemplateSrv,
+  value?: string,
+  scoped?: ScopedVars,
+  format?: string
+): string | undefined => {
+  return !!value ? t.replace(value, scoped, format) : value;
 };
 
 export const replaceVariables = (t: TemplateSrv, query: GitHubQuery, scoped: ScopedVars): GitHubQuery => {
   Object.keys(query).forEach((key) => {
     if (typeof query[key] === 'string') {
-      query = { ...query, [key]: replaceVariable(t, query[key], scoped) };
+      if (key === 'query') {
+        query = { ...query, [key]: replaceVariable(t, query[key], scoped, 'csv') };
+      } else {
+        query = { ...query, [key]: replaceVariable(t, query[key], scoped) };
+      }
     }
   });
 
   if (query.options) {
     let { options } = query;
     Object.keys(options).forEach((key) => {
       if (typeof options[key] === 'string') {
-        options = { ...options, [key]: replaceVariable(t, options[key], scoped) };
+        if (key === 'query') {
```

Comment:
looks like we are repeating ourselves here.  same as https://github.com/grafana/github-datasource/pull/162/files#diff-7b3e863a7f8b31310b0cd25e6933ac6ae774750e4399e9621f11af435172d2d2R17-R21

maybe create a function to DRY it up

Reply:
👍🏼 addressed in https://github.com/grafana/github-datasource/pull/162/commits/7c2a17d400fa07c4ad88e25019e234e1f2d28954 and added tests.

We should also aim https://github.com/grafana/github-datasource/issues/43 if we decide to refactor this further.